.github/workflows/build-test-ship.yaml

```diff
@@ -131,21 +134,25 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
-      # On push/PR, ensure service images can be built on target platforms.
-      - name: Build final images
+      # Build and upload image artifact for use in api-proctor tests.
+      - name: Build image for api-proctor tests
         uses: docker/build-push-action@v2
-        if: |
-          github.event_name == 'pull_request' ||
-          (github.event_name == 'push' &&
-           steps.check-version-change.outputs.appchange == 'false')
         with:
           context: .
           file: ./cmd/${{ matrix.service }}/Dockerfile
           cache-from: type=gha
           cache-to: type=gha,mode=max
-          platforms: linux/amd64,linux/arm64
+          outputs: type=docker,dest=${{ matrix.service }}.tar
+          tags: ${{ matrix.service }}:ci
 
-      # On release, build service images and push to ECR.
+      - name: Upload docker image artifact
```

Comment:
```suggestion
      # Since jobs run on different machines (unless you're using custom runners), we can't trivially access files produced in one job from another.
      # We have to upload the files in one job and download them in another, which we can do via GitHub
      # Artifacts, which we upload as tagged .TAR files
      - name: Upload docker image artifact
```

Reply:
> clarifying PR comments ought to be included in the file itself

Good idea! Reworded and added in ec4c8fdda385d0ec963b0f24388a9f4e96c4d0e5